src/test/java/com/iota/iri/service/validation/impl/TransactionSolidifierImplTest.java

```diff
@@ -0,0 +1,149 @@
+    private static final TemporaryFolder logFolder = new TemporaryFolder();
+    private static Tangle tangle;
+
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private static SnapshotProvider snapshotProvider;
+
+    @Mock
+    private static TransactionSolidifierImpl txSolidifier;
+
+    @Mock
+    private static TransactionRequester txRequester;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        dbFolder.create();
+        logFolder.create();
+        tangle = new Tangle();
+        tangle.addPersistenceProvider(
+                new RocksDBPersistenceProvider(
+                        dbFolder.getRoot().getAbsolutePath(), logFolder.getRoot().getAbsolutePath(),1000, Tangle.COLUMN_FAMILIES, Tangle.METADATA_COLUMN_FAMILY));
+        tangle.init();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        tangle.shutdown();
+        dbFolder.delete();
+        logFolder.delete();
+    }
+
+    @Before
+    public void setUpEach() {
+        when(snapshotProvider.getInitialSnapshot()).thenReturn(SnapshotMockUtils.createSnapshot());
+        txRequester = new TransactionRequester(tangle, snapshotProvider);
+        txSolidifier = new TransactionSolidifierImpl(tangle, snapshotProvider, txRequester);
+        txSolidifier.start();
+    }
+
+    @After
+    public void tearDownEach(){
+        txSolidifier.shutdown();
+    }
+
+
+    @Test
+    public void verifyTxIsSolid() throws Exception {
+        TransactionViewModel tx = getTxWithBranchAndTrunk();
+        assertTrue(txSolidifier.checkSolidity(tx.getHash()));
+        assertTrue(txSolidifier.checkSolidity(tx.getHash()));
+    }
+
+    @Test
+    public void verifyTxIsNotSolid() throws Exception {
+        TransactionViewModel tx = getTxWithoutBranchAndTrunk();
+        assertFalse(txSolidifier.checkSolidity(tx.getHash()));
+        assertFalse(txSolidifier.checkSolidity(tx.getHash()));
+    }
+
+    @Test
+    public void getSolidificationQueue() throws Exception{
+        TransactionViewModel tx = getTxWithBranchAndTrunk();
+        txSolidifier.addToSolidificationQueue(tx.getHash());
+
+    }
+
+    @Test
+    public void verifyTransactionIsProcessedFully() throws Exception {
+        TransactionViewModel tx = getTxWithBranchAndTrunk();
+        txSolidifier.addToSolidificationQueue(tx.getHash());
+        assertTrue(txSolidifier.getSolidificationQueue().contains(tx.getHash()));
+        //Time to process through the steps
+        Thread.sleep(1000);
+        assertTrue(txSolidifier.getBroadcastQueue().contains(tx));
+    }
+
+
+    @Test
+    public void verifyInconsistentTransactionIsNotProcessedFully() throws Exception {
+        TransactionViewModel tx = getTxWithoutBranchAndTrunk();
+        txSolidifier.addToSolidificationQueue(tx.getHash());
+        assertTrue(txSolidifier.getSolidificationQueue().contains(tx.getHash()));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit assertions should include a message](https://app.codacy.com/app/GalRogozinski/iri/pullRequest?prid=4339340)